src/main/java/org/jabref/logic/sharelatex/ShareLatexManager.java

```diff
@@ -0,0 +1,76 @@
+package org.jabref.logic.sharelatex;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+
+import org.jabref.JabRefExecutorService;
+import org.jabref.logic.exporter.BibtexDatabaseWriter;
+import org.jabref.logic.exporter.SaveException;
+import org.jabref.logic.exporter.SavePreferences;
+import org.jabref.logic.exporter.StringSaveSession;
+import org.jabref.logic.importer.ImportFormatPreferences;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.sharelatex.ShareLatexProject;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class ShareLatexManager {
+
+    private static final Log LOGGER = LogFactory.getLog(ShareLatexManager.class);
+
+    private final SharelatexConnector connector = new SharelatexConnector();
+    private final ShareLatexParser parser = new ShareLatexParser();
+
+    public String login(String server, String username, String password) throws IOException {
+        return connector.connectToServer(server, username, password);
+    }
+
+    public List<ShareLatexProject> getProjects() throws IOException {
+        if (connector.getProjects().isPresent()) {
+            return parser.getProjectFromJson(connector.getProjects().get());
+        }
+        return Collections.emptyList();
+    }
+
+    public void startWebSocketHandler(String projectID, BibDatabaseContext database, ImportFormatPreferences preferences) {
+        JabRefExecutorService.INSTANCE.executeAndWait(() -> {
+
+            try {
+                connector.startWebsocketListener(projectID, database, preferences);
+            } catch (URISyntaxException e) {
+                LOGGER.error(e);
+            }
+            registerListener(ShareLatexManager.this);
+
+        });
+    }
+
+    public void sendNewDatabaseContent(BibDatabaseContext database) {
```

Comment:
@koppor  Here the changes are sended, the bibtexdb is written into a string and then the changes are calculated